src/main/java/com/amazonaws/codedeploy/AWSCodeDeployPublisher.java

```diff
@@ -222,10 +225,21 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
             RevisionLocation revisionLocation = zipAndUpload(aws, projectName, getSourceDirectory(build.getWorkspace()), envVars);
 
             registerRevision(aws, revisionLocation);
-            String deploymentId = createDeployment(aws, revisionLocation);
+            deploymentId = createDeployment(aws, revisionLocation);
 
             success = waitForDeployment(aws, deploymentId);
 
+        } catch (InterruptedException e) {
+
+            this.logger.println("User cancelling CodeDeploy job");
+	    try {
+	      this.logger.println(stopDeployment(aws, deploymentId));
+	    } catch (Exception ex) {
+                this.logger.println(e.getMessage());
+		e.printStackTrace(this.logger);
+	    }
+	    success = false;
```

Comment:
If the stopDeployment fails because the deployment completed successfully, we shouldn't fail the job, but actually mark it as successful.